SequenceReplay/SequenceReplay.py

```diff
@@ -255,95 +256,103 @@ def setParameterNode(self, inputParameterNode: Optional[SequenceReplayParameterN
-        for node in slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode"):
-            if node.GetID() == sequenceBrowserNodeId:
-                node.SetPlaybackActive(True)
+
+        # Toggle playback state
+        if sequenceBrowserNodeId in self.playbackState:  # If the node is currently playing
+            if self.playbackState[sequenceBrowserNodeId]:
+                sequenceBrowserNode.SetPlaybackActive(False)  # stop
+                self.playbackState[sequenceBrowserNodeId] = False
+                self.buttonState[sequenceBrowserNodeId].setText("Play " + sequenceBrowserNode.GetName())
             else:
+                sequenceBrowserNode.SetPlaybackActive(True)  # If the node is not currently playing
+                self.playbackState[sequenceBrowserNodeId] = True  # start
+                self.buttonState[sequenceBrowserNodeId].setText("Stop " + sequenceBrowserNode.GetName())
+        else:  # not already in the playbackState dictionary
+            sequenceBrowserNode.SetPlaybackActive(True)
+            self.playbackState[sequenceBrowserNodeId] = True
+            self.buttonState[sequenceBrowserNodeId].setText("Stop " + sequenceBrowserNode.GetName())
+
+        # Ensure other sequence browser nodes are stopped
+        for node in slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode"):
+            if node.GetID() != sequenceBrowserNodeId:
                 node.SetPlaybackActive(False)
-        
+                self.playbackState[node.GetID()] = False
+                self.buttonState[node.GetID()].setText("Play " + node.GetName())
+
     def _updateGui(self, caller=None, event=None) -> None:
-        """Update GUI based on parameter node and mrml scene state."""
-        if self._addingNodes:
-            return
-        
-        if self._importingScene:
+        if self._addingNodes or self._importingScene:
             return
-        
-        # Find all sequence browser nodes and create a button for each of them
-        
-        sequenceBrowserNodes = slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode")
-        
+
+        self._updateSequenceBrowserButtons()
+        self._updateTransformableNodeCheckboxes()
+
+    def _updateSequenceBrowserButtons(self):
         sequenceBrowsersLayout = self.ui.sequenceBrowsersCollapsibleButton.layout()
-        
-        # Remove all buttons from the layout
-        for i in reversed(range(sequenceBrowsersLayout.count())):
-            widget = sequenceBrowsersLayout.itemAt(i).widget()
-            widget.setParent(None)
-            widget.deleteLater()
-            
-        # Create a button for each node
+        self._clearLayout(sequenceBrowsersLayout)
+
+        sequenceBrowserNodes = slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode")
         for node in sequenceBrowserNodes:
-            button = qt.QPushButton(node.GetName())
+            button = qt.QPushButton("Play " + node.GetName())
```

Comment:
It would be better if all the button creation/deletion/updates were done here, triggered by MRML node events.